lib/checkio.cpp

```diff
@@ -87,27 +87,27 @@ void CheckIO::coutCerrMisusageError(const Token* tok, const std::string& streamN
 // fopen("","r"); fwrite(); <- write to read-only file (or vice versa)
 // fclose(); fread(); <- Use closed file
 //---------------------------------------------------------------------------
-enum OpenMode { CLOSED, READ_MODE, WRITE_MODE, RW_MODE, UNKNOWN_OM };
+enum class OpenMode { CLOSED, READ_MODE, WRITE_MODE, RW_MODE, UNKNOWN_OM };
 static OpenMode getMode(const std::string& str)
 {
     if (str.find('+', 1) != std::string::npos)
-        return RW_MODE;
+        return OpenMode::RW_MODE;
     else if (str.find('w') != std::string::npos || str.find('a') != std::string::npos)
-        return WRITE_MODE;
+        return OpenMode::WRITE_MODE;
     else if (str.find('r') != std::string::npos)
-        return READ_MODE;
-    return UNKNOWN_OM;
+        return OpenMode::READ_MODE;
+    return OpenMode::UNKNOWN_OM;
 }
 
 struct Filepointer {
     OpenMode mode;
     nonneg int mode_indent;
-    enum Operation {NONE, UNIMPORTANT, READ, WRITE, POSITIONING, OPEN, CLOSE, UNKNOWN_OP} lastOperation;
+    enum class Operation {NONE, UNIMPORTANT, READ, WRITE, POSITIONING, OPEN, CLOSE, UNKNOWN_OP} lastOperation;
```

Comment:
For information.. Refactoring `Filepointer::Operation` and `Filepointer::AppendMode` does not feel very important to me. When the enum constants are defined in a small struct they have a small well defined scope already.

Reply:
I think the added type safety is good, but i see your point.